utils.py

```diff
@@ -0,0 +1,245 @@
+import glob
+import logging
+import os
+import random
+import sys
+from itertools import product
+import math
+
+import numpy as np
+import pandas as pd
+import torch
+import yaml
+from PIL import Image
+from attrdict import AttrDict
+from tqdm import tqdm
+import json
+import pycocotools
+
+
+def read_yaml(filepath):
+    with open(filepath) as f:
+        config = yaml.load(f)
+    return AttrDict(config)
+
+
+def init_logger():
+    logger = logging.getLogger('mapping-challenge')
+    logger.setLevel(logging.INFO)
+    message_format = logging.Formatter(fmt='%(asctime)s %(name)s >>> %(message)s',
+                                       datefmt='%Y-%m-%d %H-%M-%S')
+
+    # console handler for validation info
+    ch_va = logging.StreamHandler(sys.stdout)
+    ch_va.setLevel(logging.INFO)
+
+    ch_va.setFormatter(fmt=message_format)
+
+    # add the handlers to the logger
+    logger.addHandler(ch_va)
+
+    return logger
+
+
+def get_logger():
+    return logging.getLogger('mapping-challenge')
+
+
+def decompose(labeled):
+    nr_true = labeled.max()
+    masks = []
+    for i in range(1, nr_true + 1):
+        msk = labeled.copy()
+        msk[msk != i] = 0.
+        msk[msk == i] = 255.
+        masks.append(msk)
+
+    if not masks:
+        return [labeled]
+    else:
+        return masks
+
+
+def create_submission(meta, predictions, logger, save=False, experiment_dir='./'):
+    '''
```

Comment:
@apyskir to be reformatted to the Google style (still :awesome: that you wrote docstring).